### PR TITLE
Hierarchical divisions

### PR DESCRIPTION
**Pull request by jessecoleman**
Sorry this one is so big. I went ahead and moved (almost) all the filter state into the URL as part of the refactor cause I was running into some bugs with the previous filter state.

### CODE REVIEW

File: frontend/src/app/actions.ts [hunk 1]
```diff
@@ -1,115 +1,71 @@
 import 'whatwg-fetch';
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import { Grant, SearchResponse, Service } from 'api';
+import { Grant, ServiceService as Service } from 'api';
```
Comment (h1-the-swan) [inline frontend/src/app/actions.ts:3]: Is this supposed to be `ServiceService` or `Service`?

Comment (jessecoleman) [reply frontend/src/app/actions.ts:3]: I upgraded the codegen package and it now takes an optional `service-name` that is "Service" by default. I didn't want to mess with the CLI codegen in your build scripts so I left it alone.

Comment (h1-the-swan) [reply frontend/src/app/actions.ts:3]: But this throws an error: `Attempted import error: 'ServiceService' is not exported from 'api' (imported as 'Service').`

File: frontend/scripts/generateAPI.mjs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (h1-the-swan) [inline frontend/scripts/generateAPI.mjs:1]: why was this file renamed from `.js` to `.mjs`?

**Review by h1-the-swan — APPROVED** [review]
Great! Looks like things are working.
